### PR TITLE
Add Calendar components page to styleguide

### PR DESCRIPTION
**Pull request by farazcsk**

### CODE REVIEW

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+              specimenContainer: m.par,
+            }}
+            code={dedent`
+              <CalendarMonth />
+            `}
+          >
+            <CalendarMonth />
+          </Specimen>
+
+          <H level={4} className={m.mtLgIi}>External Dates</H>
```
Comment (NGMarmaduke) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: Eternal Dates is a weird title - Out of Range may work better but still not keen.

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+          <T elm="p" className={m.mtr}>
+            By default the <C>CalendarMonth</C> component hides dates outside of the current month.
+            To overide this behaviour use the <C>showOutOfRange</C> prop show below. The out of
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: Typos:

- [ ] `overide` -> `override`
- [ ] `show` -> `shown`

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+              columnHeadingProps={{
+                format: 'dddd',
+              }}
+            />
+          </Specimen>
+
+          <H level={3} className={m.mtLgIi}>Day Picker</H>
+          <T elm="p" className={m.mtr}>
+            Like the calendar month, but allows a day to be selected when clicked or touched this is
+            indicated by a filled square. Using the left and right chevrons you can navigate
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: > Like the calendar month, but allows a day to be selected when clicked or touched this is indicated by a filled square.

I don't know if it's me but this sentence sounds weird. 
Couldn't it be something like:

> Like the calendar month, but allows a day to be selected when clicked or touched; the selected day is indicated by a filled square.

?

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+              onInteraction={this.handleDayPickerInteraction}
+              onMonthChange={this.handleDayPickerMonthChange}
+              dayProps={{ getDayState: this.getDayState }}
+            />
+          </Specimen>
+
+          <H level={3} className={m.mtLgIi}>Day Range Picker</H>
+          <T elm="p" className={m.mtr}>
+            Clicking a date selects a start date moving the mouse into the future indicates the
+            range using a row or block of filled squares. Clicking again selects the end date.
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: Also this one sounds a bit weird, like punctuation is missing or something 👀

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+              />
+            `}
+          >
+            <DayRangePicker
+              startDate={dayRangePicker.startDate}
+              endDate={dayRangePicker.endDate}
+              selectDate={dayRangePicker.selectDate}
+              month={dayRangePicker.month}
+              onInteraction={this.handleDayRangeInteraction}
+              onMonthChange={this.handleDayRangeMonthChange}
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: I tried this one out but it's not working as expected. Changing month and then clicking on a day of that month brings me back to the selection I already made in the previous month.

Comment (farazcsk) [reply styleguide/screens/Patterns/Calendar/Calendar.js]: I agree this behaviour could be improved, let's address this separately.

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+  handleDayPickerInteraction = (e, date) => {
+    this.setState(state => ({
+      dayPicker: {
+        ...state.dayPicker,
+        date,
+      },
+    }));
+  };
+
+  handleDayPickerInteraction = (e, date) => {
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: Duplicated function here ⬆️

File: styleguide/screens/Patterns/Calendar/Calendar.js [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+            <CalendarMonth
+              columnHeadingProps={{
+                format: 'dddd',
+              }}
+            />
+          </Specimen>
+
+          <H level={3} className={m.mtLgIi}>Day Picker</H>
+          <T elm="p" className={m.mtr}>
+            Like <C>calendarMonth</C>, but allows a day to be selected when clicked or touched;
```
Comment (desmondhume) [inline styleguide/screens/Patterns/Calendar/Calendar.js]: `calendarMonth` -> `CalendarMonth`

**Pull request merged**